lib/src/dialog/Dialog.tsx

```diff
@@ -27,92 +26,124 @@ const DxcDialog = ({
-      <DialogContainer
-        open={true}
-        isCloseVisible={isCloseVisible}
-        onClose={handleOverlayClick}
-        overlay={overlay}
-        padding={padding}
-      >
-        {isCloseVisible && (
-          <CloseIconContainer onClick={handleClose} tabIndex={tabIndex}>
-            <CloseIcon
-              xmlns="http://www.w3.org/2000/svg"
-              width="24"
-              height="24"
-              viewBox="0 0 24 24"
-              fill="currentColor"
-            >
-              <path d="M0 0h24v24H0V0z" fill="none" />
-              <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z" />
-            </CloseIcon>
-          </CloseIconContainer>
-        )}
-        <Children>
-          <BackgroundColorProvider color={colorsTheme.dialog.backgroundColor}>{children}</BackgroundColorProvider>
-        </Children>
+      <BodyStyle />
+      <DialogContainer role="presentation">
+        {overlay && <Overlay onClick={handleOverlayClick} />}
+        <Dialog role="dialog" aria-modal={overlay} isCloseVisible={isCloseVisible}>
+          <Children padding={padding}>
+            <BackgroundColorProvider color={colorsTheme.dialog.backgroundColor}>{children}</BackgroundColorProvider>
+          </Children>
+          {isCloseVisible && (
+            <CloseIconContainer onClick={handleClose} onKeyDown={handleOnKeyDown} tabIndex={tabIndex} ref={ref}>
+              <CloseIcon
+                xmlns="http://www.w3.org/2000/svg"
+                width="24"
+                height="24"
+                viewBox="0 0 24 24"
+                fill="currentColor"
+              >
+                <path d="M0 0h24v24H0V0z" fill="none" />
+                <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z" />
+              </CloseIcon>
+            </CloseIconContainer>
+          )}
+        </Dialog>
       </DialogContainer>
     </ThemeProvider>
   );
 };
 
-const DialogContainer = styled(Dialog)`
-  overflow: unset;
+const BodyStyle = createGlobalStyle`
+  body {
+    overflow: hidden;
+  }
+`;
+
+const DialogContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  position: fixed;
+  inset: 0px;
+  height: 100%;
+  z-index: 1300;
+`;
+
+const Overlay = styled.div`
+  position: fixed;
+  inset: 0px;
+  height: 100%;
+  background-color: ${(props) => props.theme.overlayColor};
+  opacity: ${(props) => props.theme.overlayOpacity};
+`;
+
+const Dialog = styled.div<{ isCloseVisible?: boolean }>`
+  display: flex;
+  justify-content: space-between;
+  z-index: 1300;
+  background-color: ${(props) => props.theme.backgroundColor};
   font-family: ${(props) => props.theme.fontFamily};
+  font-size: ${(props) => props.theme.fontSize};
+  font-weight: ${(props) => props.theme.fontWeight};
+  min-height: ${(props) => (props.isCloseVisible ? "72px" : "")};
```

Comment:
You could refactor this line with `${(props) => (props.isCloseVisible && "min-height: 72px;")}`.